clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12969,6 +12969,8 @@ def err_free_function_first_occurrence_missing_attr: Error<
   "the first occurrence of SYCL kernel free function should be declared with 'sycl-nd-range-kernel' or 'sycl-single-task-kernel' compile time properties">;
 def err_free_function_class_method : Error<
   "%select{static |}0class method cannot be used to define a SYCL kernel free function kernel">;
+def err_free_function_virtual_arg : Error<
+  "argument type '%0' virtually inherited from base class is not supported as a SYCL kernel free function kernel arguments">;
```

Comment:
A couple of NITs:
1. This is not specific for free function kernels, so let's make a generic message that can be reused.
2. I think we can actually point to the base class to make it more clear.
```suggestion
  "argument type '%0' virtually inherited from base class `%1` is not supported as a SYCL kernel argument">;
```